version: '3'

services:
  django:
    image: django
    build: .
    user: ${DJANGO_USER:-yeboyebo}
    restart: always
    working_dir: /src/app/
    environment:
      - TZ=Europe/Amsterdam
      - LC_ALL=C.UTF-8
      - LANG=C.UTF-8
      - LANGUAGE=C.UTF-8
    env_file:
      - ./.env
    ports:
      - "${PORT}:8000"
    extra_hosts:
      - "dbhost:${DBHOST}"
    volumes:
      - .:/src
      - ${PINEBOODIR}:/pineboo/pineboo
      - ${MODULESDIR}:/pineboo/modules
      - ~/.config/Eneboo/PinebooSettings.ini:/home/yeboyebo/.config/Eneboo/PinebooSettings.ini
      - /run/cups/cups.sock:/run/cups/cups.sock
      - /etc/cups/cupsd.conf:/etc/cups/cupsd.conf
<<<<<<< HEAD
      - ${IMAGESVOL:-/tmp}:/static/images
      - ${EXTERNAL_MODULES:-/tmp}:/external
=======
      - ${IMAGESVOL:-./app/assets/dist/img/}:/static/images
      - ${EXTERNAL_MODULES:-./app/controllers/api/}:/external
>>>>>>> 4abc11967ffa4f1e7f606fe62fa1f33646a277a0
    command: python3 manage.py runserver 0.0.0.0:8000 --noreload
    networks:
      app_network:

networks:
  app_network:
    driver: bridge
    ipam:
      config:
        - subnet: 172.${DOCKER_IP}.0.0/16

